src/conn.rs

```diff
@@ -615,6 +659,108 @@ impl LdapConnAsync {
 
     async fn turn(mut self, mode: LoopMode) -> Result<Self> {
         loop {
+            #[cfg(any(feature = "tls-native", feature = "tls-rustls"))]
+            tokio::select! {
```

Comment:
This part is not great, because it duplicates the whole `tokio::select` block, see https://github.com/tokio-rs/tokio/issues/3974

We could do private functions for `id_scrub_rx`, ... so the duplication is smaller, i just wanted to ask how you feel about this before doing private functions

Reply:
I don't think it's necessary to duplicate the whole of `select!` if the channel is more generic (i.e., not `tx_cert` but let's say `tx_misc`) and the transferred request is an enum whose variants can be cfg-gated _inside_ the `select!` branch code block. This means that the connection struct gains an extra channel receiver, and the `Ldap` struct a sender, even when no TLS is configured, but that's not a huge price to pay for much cleaner code.

Reply:
Yeah that makes sense. I could implement this tomorrow